artifact_loader/zip_loader.go

```diff
@@ -0,0 +1,72 @@
+package artifact_loader
+
+import (
+	"archive/zip"
+	"context"
+	"fmt"
+	"io"
+	"log/slog"
+
+	"github.com/turbot/tailpipe-plugin-sdk/types"
+)
+
+const ZipLoaderIdentifier = "zip_loader"
+
+// ZipLoader is a Loader that extracts a zip file and returns all the content
+type ZipLoader struct {
+}
+
+func NewZipLoader() Loader {
+	return &ZipLoader{}
+}
+
+func (z ZipLoader) Identifier() string {
+	return ZipLoaderIdentifier
+}
+
+// Load implements Loader
+// Extracts an object from a zip file
+func (z ZipLoader) Load(ctx context.Context, info *types.DownloadedArtifactInfo, dataChan chan *types.RowData) error {
+	slog.Debug("ZipLoader Load", "path", info.LocalName)
+	inputPath := info.LocalName
+
+	// Open the zip file
+	zipFile, err := zip.OpenReader(inputPath)
+	if err != nil {
+		return fmt.Errorf("error opening %s: %w", inputPath, err)
+	}
+	defer zipFile.Close()
+
+	// Check if the zip file has any files
+	if len(zipFile.File) == 0 {
+		return fmt.Errorf("zip file %s is empty", inputPath)
+	}
+
+	// For now, we'll just read the first file in the zip archive
```

Comment:
is this a reasonable shortcut to take or should we just handle all files in the archive?

Reply:
I believe we have a similar assumption in the gzip loaders that the compressed file contains a single entity?

Reply:
As discussed, we now return an error if multiple files are inside a zip archive - unfortunately, we can't determine the same for gz archives - instead it appears the go package will just silently ignore all but the first file - this is due to default behaviour of gz being single file (tar.gz for multiple file).